p2pool/src/sharechain/p2chain.rs

```diff
@@ -514,20 +513,22 @@ impl<T: BlockCache> P2Chain<T> {
                 new_tip.set_new_tip(hash, new_block_height);
                 // we need to reorg the chain
                 // lets start by resetting the lwma
-                self.lwma = LinearWeightedMovingAverage::new(DIFFICULTY_ADJUSTMENT_WINDOW, self.block_time)
+                let mut lwma = LinearWeightedMovingAverage::new(DIFFICULTY_ADJUSTMENT_WINDOW, self.block_time)
                     .expect("Failed to create LWMA");
-                self.lwma.add_front(block.timestamp, block.target_difficulty);
+                lwma.add_front(block.timestamp, block.target_difficulty);
                 let chain_height = self
                     .level_at_height(block.height)
-                    .ok_or(ShareChainError::BlockLevelNotFound)?;
+                    .ok_or(ShareChainError::BlockLevelNotFound)?; // Do we need a panic here?
```

Comment:
I think we should panic here, and probably delete the lmdb database as well,
If the block was verified, it needs to have it parents. Now we cannot find the parent anymore.